pkgs/development/tools/haskell/haskell-language-server/default.nix

```diff
@@ -0,0 +1,67 @@
+{ mkDerivation, fetchgit
```

Comment:
What do you mean by "update" ? 
My understanding is that thoses derivations depend on their github repositories, so updating means changing current revision. Anything else seems to be the "common" haskell nixpkgs infrastructure.

Reply:
Maybe I'm not understanding the structure here.

So you've generated `pkgs/development/tools/haskell/haskell-language-server/default.nix` and `pkgs/development/tools/haskell/haskell-language-server/hls-ghcide.nix` by hand with `cabal2nix`?

Ideally there would be an easy way to regenerate these files with `cabal2nix`, so that someone could easily update it if a new version of one of these packages was released to github.

Without having a (semi-?)automated script, a potential future contributor would just have to know that those files were generated by `cabal2nix`, and how to run `cabal2nix` again to generate them.

Reply:
Yes, I cloned the github repos and used cabal2nix to get those, but I also had to edit it a bit (e.g. change src, add git dependency)

I believe that those packages will be updated by 
1. Changing git rev
2. Manual trial/error process as any other nixpkgs package

Though the forked `ghcide` might disappear.

HLS is quite unstable, so I'm not ready to automate it (yet).

I will document how those files where written at the top of each

Reply:
As long as you're using cabal2nix and the Haskell architecture in nixpkgs, it should be easy to automate the creation and updates of these two derivations.

There is often no reason to edit derivations from `cabal2nix` by hand (except as overrides in configuration-common.nix).

Reply:
cabal2nix does miss the git dependency and the preCheck hook:

https://github.com/NixOS/nixpkgs/blob/72e5c024af9af386886d1a653189b9b0629254a5/pkgs/development/tools/haskell/haskell-language-server/default.nix#L57-L60

Reply:
> cabal2nix does miss the git dependency and the preCheck hook

Ah, yeah, so these two things should be added in the `configuration-common.nix` file (if possible).